src/Microsoft.IdentityModel.Logging/LogHelper.cs

```diff
@@ -339,16 +339,26 @@ public static void LogWarning(string message, params object[] args)
             if (Logger.IsEnabled(eventLogLevel))
                 Logger.Log(WriteEntry(eventLogLevel, innerException, message, null));
 
-            if (innerException != null) 
+            T? value;
+            if (innerException != null)
+            {
                 if (string.IsNullOrEmpty(argumentName))
-                    return (T)Activator.CreateInstance(typeof(T), message, innerException);
+                    value = Activator.CreateInstance(typeof(T), message, innerException) as T ;
                 else
-                    return (T)Activator.CreateInstance(typeof(T), argumentName, message, innerException);
+                    value = Activator.CreateInstance(typeof(T), argumentName, message, innerException) as T;
+            }
             else
+            {
                 if (string.IsNullOrEmpty(argumentName))
-                    return (T)Activator.CreateInstance(typeof(T), message);
+                    value = Activator.CreateInstance(typeof(T), message) as T;
                 else
-                    return (T)Activator.CreateInstance(typeof(T), argumentName, message);
+                    value = Activator.CreateInstance(typeof(T), argumentName, message) as T;
+            }
+
+            if(value is null)
+                throw new ArgumentException("Newly created Exception is null");
```

Comment:
we need a better error message, this isn't actionable.

Reply:
@keegan-caruso I can elaborate more on this error message however, because 'value' is created local to this method, there isn't a specific param I can reference to as part of the ArgumentException. Do you have any suggestions to improve on this?

Reply:
either build up the message based on the path taken above or list out all of the params even if they are potentially null/unused?

Reply:
at a minimum we could at least call out the type?

Reply:
I am not sure which one will make the most sense in here. @jennyf19 @jmprieur do you any suggestions on improving this message?